File: Lab1/Temperature/Fahrenheit.cs
namespace Temperature;

class Fahrenheit : Temperature
{
    public Fahrenheit(double degrees) : base(degrees)
    {
        if (degrees < -459.67) throw new ArgumentException("Absolute zero on Fahrenheit is -459.67°F.");
    }

    public Celsius ToCelsius()
    {
        return new Celsius((this.value - 32) / 1.8);
    }

    public Kelvin ToKelvin()
    {
        return ToCelsius().ToKelvin();
    }
}

File: Lab1/src/Temperature/Fahrenheit.java
package SoftwareEngineering.Lab1.Temperature;

public class Fahrenheit extends BaseTemperature {
    public Fahrenheit(double temperature) {
        // TODO Absolute zero on Fahrenheit is -459.67°F
        super(temperature);
    }

    public Celsius toCelsius() {
        return new Celsius((temperature - 32) / 1.8);
    }

    public Fahrenheit toFahrenheit() {
        return new Fahrenheit(temperature);
    }

    public Kelvin toKelvin() {
        return toCelsius().toKelvin();
    }
}